src/game/mod.rs

```diff
@@ -23,12 +27,26 @@ impl Game {
                 Input::Render(args) => {
                     opengl.draw(args.viewport(), |context, graphics| {
                         clear(color::BLACK, graphics);
+                        for bullet in &self.bullets {
+                            bullet.draw(context, graphics);
+                        }
                         self.player.draw(context, graphics);
                     });
                 }
 
                 Input::Update(args) => {
                     self.player.update(args);
+                    if self.player.actions.is_shooting && self.player.can_shoot() {
```

Comment:
We could change player.is_shooting() to player.should_shoot() and have it check both the action AND the cooldown, to simplify this condition.